agent/agent.go

```diff
@@ -14,18 +14,25 @@ import (
 	"github.com/influxdata/telegraf/config"
 	"github.com/influxdata/telegraf/internal"
 	"github.com/influxdata/telegraf/models"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/outputs"
 	"github.com/influxdata/telegraf/plugins/serializers/influx"
 )
 
 // Agent runs a set of plugins.
 type Agent struct {
-	Config *config.Config
+	Config 			*config.Config
+	iu     			*inputUnit
+	wg     			sync.WaitGroup
+	inputStatus 	map[string](chan string)
```

Comment:
Since the only purpose of this channel is to stop a plugin, instead of sending something like "STOP" through the channel, you could just make it a generic channel that sends a struct. (This is what you normally do if you're using the channel to signal one type of thing.

We probably also want a way to shutdown outputs, or at least build this so that it can. So it might be a good idea to have a multi layered map like
map[string][string]chan struct{}

where the first layer is either ```INPUT```, ```OUTPUT``` and the second layer is the name of the plugin.

So for example, to shut down cpu as an input:
```
pluginShutdowns map[string][string]chan struct{}
pluginShutdowns["input"]["cpu"] <- struct{}{}
```

And then in the select case within the ```gatherLoop```, you just check if anything comes in on that channel (don't need to check if it equals "STOP", because it will never be anything but a stop signal.

Let me know what you think of this